pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -139,11 +139,11 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pa *pav1alpha1.PodAutosc
 	// If SKS is not ready — ensure we're not becoming ready.
 	// TODO: see if we can perhaps propagate the SKS state to computing active status.
 	if !sks.IsReady() {
-		logger.Debug("SKS is not ready, marking dependencies not ready")
+		logger.Debug("SKS is not ready, marking SKS status not ready")
 		ready = 0
-		pa.Status.MarkSKSNotReady()
+		pa.Status.MarkSKSNotReady(sks.Status.GetCondition(nv1alpha1.ServerlessServiceConditionReady).Message)
```

Comment:
This can return nil, so I'd guess this will make our controllers crash loop 😬

Reply:
This one cannot: we get here only if PrivateServiceName is set, which means SKS reconciled this at least once, which means InitializeConditions has been invoked and sks updated the status.

Reply:
i see the `sks.Status.PrivateServiceName` guard, which ensures things have been initialized to reach this point.